src/chess/piece/Rook.java

```diff
@@ -18,7 +18,12 @@ public boolean isValidMove(Position newPosition) {
     if (!super.isValidMove(newPosition)) {
       return false;
     }
-
-    return true;
+    // If we passed the first test then check for the specific rook movement
+    if (newPosition.getCol() == this.position.getCol()
+        && newPosition.getRow() == this.position.getRow()) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
You should arrange code format, on other parts, you actually didin't put else statement when the value is returned within if statement